app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt

```diff
@@ -245,6 +290,82 @@ fun hikeDetailsMap(hike: DetailedHike): MapView {
     }
   }
 
+  // Create state values that we can actually observe in the LaunchedEffect
+  // We keep our StateFlows for debouncing
+  val boundingBoxState = remember { MutableStateFlow(mapView.boundingBox) }
+  val zoomLevelState = remember { MutableStateFlow(mapView.zoomLevelDouble) }
+
+  // Update the map listener to just update the StateFlows
+  mapView.addMapListener(
+      object : MapListener {
+        override fun onScroll(event: ScrollEvent?): Boolean {
+          // On a Scroll event the boundingBox will change
+          event?.let {
+            val newBoundingBox = mapView.boundingBox
+            if (newBoundingBox != boundingBoxState.value) {
+              boundingBoxState.value = newBoundingBox
+            }
+          }
+          return true
+        }
+
+        override fun onZoom(event: ZoomEvent?): Boolean {
+          event?.let {
+            val newZoomLevel = mapView.zoomLevelDouble
+            if (newZoomLevel != zoomLevelState.value) {
+              zoomLevelState.value = newZoomLevel
+            }
+          }
+          return true
+        }
+      })
+
+  LaunchedEffect(Unit) {
+
+    // Create our combined flow
+    val combinedFlow =
+        combine(
+            boundingBoxState.debounce(HikeDetailScreen.DEBOUNCE_DURATION),
+            zoomLevelState.debounce(HikeDetailScreen.DEBOUNCE_DURATION)) { boundingBox, zoomLevel ->
+              // Return a pair of our values
+              boundingBox to zoomLevel
+            }
+
+    // Collect the flow with proper error handling
+    try {
+      combinedFlow.collect { (boundingBox, zoomLevel) ->
+        // Only proceed if the map is still valid
+        if (mapView.repository != null) {
+          facilitiesViewModel.filterFacilitiesForDisplay(
+              facilities.value ?: emptyList(),
+              bounds = boundingBox,
+              zoomLevel = zoomLevel,
+              hikeRoute = hike,
+              onSuccess = { newFacilities ->
+                // Double check map validity before updates
+                if (mapView.repository != null) {
+                  MapUtils.clearFacilities(mapView)
+                  if (newFacilities.isNotEmpty()) {
+                    MapUtils.displayFacilities(newFacilities, mapView, context)
+                  }
+                }
+              },
+              onNoFacilitiesForState = {
+                if (mapView.repository != null) {
+                  MapUtils.clearFacilities(mapView)
+                }
+              })
+        }
+      }
+    } catch (e: CancellationException) {
+      // Rethrow cancellation exceptions to allow proper coroutine cancellation
```

Comment:
`CancellationException` is a special exception in Kotlin coroutines that signals normal cancellation (like when a composable is disposed). I refactored it since in our case it isn't really necessary to throw it, since from what I understand Compose handles coroutines cancellation.